### ISSUE TITLE
Running Traefik 2

### ISSUE THREAD

**Issue by SGTGunner** [EdyTheCow/pterodactyl-docker#1]
I was wondering if you have had a chance to do this with Traefik 2. As the labels are very different?

Thanks in Advance.

Cheers,

SGTGunner

**Comment by EdyTheCow** [comment]
I didn't realize anyone actually wanted to use this! I haven't had the chance to update to Traefik 2 yet, but it's definitely on my todo list. I'll update to Traefik 2 and add more complete documentation when I get a chance!

**Comment by SGTGunner** [comment]
Very cool, thanks for the reply too. I would be happy to help out if need be and be a contributor.

**Comment by Brend-Smits** [comment]
I would actually be interested in this as well. 
Been looking to migrate my current pterodactyl to be containerized with traefik as a proxy.

**Comment by EdyTheCow** [comment]
Almost forgot about this, I'll upgrade to latest traefik and update the documentation by tonight.

**Comment by Brend-Smits** [comment]
> Almost forgot about this, I'll upgrade to latest traefik and update the documentation by tonight.

That would be amazing. Do you have any experience migrating from an existing pterodactyl installation to one with docker/traefik? Do you foresee any issues?

**Comment by EdyTheCow** [comment]
> > Almost forgot about this, I'll upgrade to latest traefik and update the documentation by tonight.
> 
> That would be amazing. Do you have any experience migrating from an existing pterodactyl installation to one with docker/traefik? Do you foresee any issues?

I've actually never ran pterodactyl outside docker before! I've been running everything in containers for past few years. But I don't see it being an issue if you just move the correct data directories to wherever you set the volumes.

**Comment by Brend-Smits** [comment]
> I've actually never ran pterodactyl outside docker before! I've been running everything in containers for past few years. But I don't see it being an issue if you just move the correct data directories to wherever you set the volumes.

Sounds good, when Version 2 is finished, I will be giving it a shot. 
If you need help converting or reviewing version 2 PR, let me know.

**Comment by SGTGunner** [comment]
It looks like version 2.2 is now done and in master branch. qq -- have you tried to send SMTP mail from the panel docker container? Do I need to configure support around TCP port 25 at the Traefik level?

**Comment by EdyTheCow** [comment]
> It looks like version 2.2 is now done and in master branch. qq -- have you tried to send SMTP mail from the panel docker container? Do I need to configure support around TCP port 25 at the Traefik level?

I haven't setup or tested anything yet, just initial clean up. I've had my hands full with other projects, this is going to take a bit longer than I expected. I'll get around to this in next few days and let you know!

**Comment by Brend-Smits** [comment]
Appreciate your work @EdyTheCow , thanks a lot for the work and making this project open source.

**Comment by EdyTheCow** [comment]
@Brend-Smits @Brend-Smits I've updated it to Traefik v2. Was able to successfully set it up using the guide provided in readme file. As of right now make sure you clone repository into /srv/docker/ or change the daemon data volume paths in daemon's container manually. I'll add extra variables to make it more flexible later. This is just what I use to organize daemon's data better.

**Comment by Brend-Smits** [comment]
Awesome news, I will be trying this out in the weekend. Will keep you posted!

**Comment by SGTGunner** [comment]
Yes, this is awesome news. Does it support using SMTP out from the panel for email?

**Comment by EdyTheCow** [comment]
> Yes, this is awesome news. Does it support using SMTP out from the panel for email?

I haven't looked at that yet, from what I gathered you need to set mail variable from mail to smtp in order to enable those options inside the panel. I'll look into it later today!

**Comment by EdyTheCow** [comment]
> > Yes, this is awesome news. Does it support using SMTP out from the panel for email?
> 
> I haven't looked at that yet, from what I gathered you need to set mail variable from mail to smtp in order to enable those options inside the panel. I'll look into it later today!

Just tested it, change MAIL_DRIVER from mail to smtp and you'll gain access to smtp options in the admin control panel.

**Comment by EdyTheCow** [comment]
I've added more variable options if you wish to use a different data directory, also updated the readme to better explain the setup.

**Comment by Brend-Smits** [comment]
I'm trying this out now, will let you know how it goes.

**Comment by SGTGunner** [comment]
I will be trying it out later tonight my time. Very cool. Thank you.

**Comment by Brend-Smits** [comment]
Everything is running fine until I try to create a server and it fails installing with the following error: https://gist.github.com/Brend-Smits/a9f37eff0eb1766890b01139ae4673fa 

I will reinstall and start from scratch to see if it's something in my configuration specifically.

**Comment by EdyTheCow** [comment]
> Everything is running fine until I try to create a server and it fails installing with the following error: https://gist.github.com/Brend-Smits/a9f37eff0eb1766890b01139ae4673fa
> 
> I will reinstall and start from scratch to see if it's something in my configuration specifically.
> 
> Edit: Looks like it was my setup, it works fine now. I will do some more testing and look into migration my network.

I've changed the path from daemon/daemon-data to daemon/data in latest commit. So that might of been the issue! Glad to hear you got it working!

**Comment by Brend-Smits** [comment]
I have successfully migrated my network to dockerized pterodactyl. It wasn't painless, there were a few issues with default networking from pterodactyl that needed switching to the one that docker-compose was using. After switching those around and changing some interfaces/connection strings. I was able to get it working 👍 
Great work, thanks a lot for your contributions. 

This issue can be closed now in my opinion.
